_meeting-notes/2019-12-03.md

```diff
@@ -0,0 +1,168 @@
+            - B) Wait until TestGround has a better DX (Dashboard, TestGround as a Service, full Network config features)
+        - On the first camp A), we can bring people to start writting Test Plans
+            - Data Point: We've railled in Vyzo, Dirk, Adin and Alex to start writting Test Plans. Dirk & Vyzo are the most advanced so far but still far
+- @jimpick - to @raulk: i've been asked to redline the spec document to show out-of-date content. What's the best way to sync it to the current state? Is this urgent?
+    - Should we:
+        - A) Declare a "Historical Document"
+        - B) Redline the spec
+    - Alternatively: C) Rewrite the spec (AI: Raul to rewrite the spec, not this week but possibly next week)
+    - Another option (in parallel) Creating mini specs, named "Design Blueprints". First one is "Test Ensembles"
+- @stebalien - to @raulk: what's the status of the DHT test refactor. The test is currently broken but I don't want to spend time fixing it if we're going to rewrite it.
+    - AI: Steven will finish the DHT Refactor PR
+- ASKs
+    - @daviddias - ASK: Please use Kanban board to communicate effectively the state of your issues/PRs. I keep seeing PRs in the "New Issues" column and wondering if you want review or not. If you haven't, read: https://github.com/ipfs/testground/blob/master/docs/HOW_WE_WORK.md
+        -~~ AI: make this discoverable! (David)~~ (moved to Assignments, below)
+    - @raulk - ASK: to @stebalien: could you write a design blueprint for traffic shaping? Let's make the current behaviour the starting point. Further enhancements / changes would be patches on top of that doc.
+        - ~~YES! AI: Steven said he can do it. <- push it to the docs folder~~ (moved to Assignments, below)
+
+
+## :dart: Next
+
+~~### OKRs for the Quarter~~ #1 Goal: Unblock go-ipfs 0.5.0
+~~https://raw.githubusercontent.com/ipfs/testground/master/docs/img/Q4OKRs.png~~
+~~### Top 3 things to get done this week~~
+
+
+
+We are putting aside the OKRs for the time being and instead **focus on the #1 Goal: Unblock go-ipfs 0.5.0**. This goal has a very clear list of things that need to ship:
+
+- [ ] Instrument go-libp2p so that the DHT operations are observable and traced in a Distributed Context (owner: @raulk)
+- [ ] Add support to Traffic Shaping & Network Configuration to TestGround (owner: @Stebalien)
+- [ ] Expand the TestGround Infrastructure so that it can run multiple thousands of nodes (owner: @nonsense)
+- [ ] Creation of a TestGround Dashboard that can aid the developers of go-ipfs & go-libp2p to grok the results outputted by every node in a TestGround run (owner: @daviddias)
+- [ ] Keep on iterating the DHT Test Plan (@owner: @daviddias)
+- [ ] Design blueprint for multi-version tests (aka test ensembles) (owner: @raulk)
+
+Full list can be at https://github.com/ipfs/testground/issues/196
+
+
+### Assignments
+
+- ==@raulk==
```

Comment:
@raulk for your week assignments, consider the following asks from previous Weekly's

- Outline a plan that describes how the libp2p instrumentation will happen, how will it be built, what is the first experiment to be made and if possible, list who will own each part
- Revisit and refactor the SPEC to get it up to date
- Create a guilde line around Error Handling Patterns (ref: https://github.com/ipfs/testground/pull/78/files#r337779827)
- How should one write a test for when it is about ingesting multiple TBs of data (ref: https://github.com/ipfs/testground/pull/69#discussion_r337443200?)

Reply:
I can commit to 1 and 3 this week (assuming our weeks go Tue to Tue), possibly gathering some thoughts for 4 too, and initiating 1.

Reply:
Added those assignments in https://github.com/ipfs/testground/pull/203/commits/6a10cf415e624985a55790e1c1c0156d1875c28b